------------------------------------------------------------
revno: 14099
revision-id: squid3@treenet.co.nz-20161015042024-jagzafukd2t6gcr0
parent: squid3@treenet.co.nz-20161009195739-pcju9hl8vqwijt26
author: Alex Rousskov <rousskov@measurement-factory.com>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Sat 2016-10-15 17:20:24 +1300
message:
  Fix build with eCAP but without ICAP support.
  
  That is, when ./configured with --enable-ecap --disable-icap-client.
  
  AccessLogEntry::icap requires ICAP_CLIENT, not just USE_ADAPTATION.
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20161015042024-jagzafukd2t6gcr0
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: 4cd2e7bf4e2be0acd252963afc107537b17450fc
# timestamp: 2016-10-15 04:52:07 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20161009195739-\
#   pcju9hl8vqwijt26
# 
# Begin patch
=== modified file 'src/format/Format.cc'
--- src/format/Format.cc	2016-09-16 11:53:28 +0000
+++ src/format/Format.cc	2016-10-15 04:20:24 +0000
@@ -318,7 +318,7 @@
 actualReplyHeader(const AccessLogEntry::Pointer &al)
 {
     const HttpMsg *msg = al->reply;
-#if USE_ADAPTATION
+#if ICAP_CLIENT
     // al->icap.reqMethod is methodNone in access.log context
     if (!msg && al->icap.reqMethod == Adaptation::methodReqmod)
         msg = al->adapted_request;
@@ -331,7 +331,7 @@
 static const HttpMsg *
 actualRequestHeader(const AccessLogEntry::Pointer &al)
 {
-#if USE_ADAPTATION
+#if ICAP_CLIENT
     // al->icap.reqMethod is methodNone in access.log context
     if (al->icap.reqMethod == Adaptation::methodRespmod) {
         // XXX: for now AccessLogEntry lacks virgin response headers
@@ -819,7 +819,7 @@
         break;
 
         case LFT_REQUEST_ALL_HEADERS:
-#if USE_ADAPTATION
+#if ICAP_CLIENT
             if (al->icap.reqMethod == Adaptation::methodRespmod) {
                 // XXX: since AccessLogEntry::Headers lacks virgin response
                 // headers, do nothing for now
@@ -843,7 +843,7 @@
 
         case LFT_REPLY_ALL_HEADERS:
             out = al->headers.reply;
-#if USE_ADAPTATION
+#if ICAP_CLIENT
             if (!out && al->icap.reqMethod == Adaptation::methodReqmod)
                 out = al->headers.adapted_request;
 #endif

